source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+        , n_cells_y(Utilities::pow(2, refinements) * n_subdivision_y)
+        , N_points_on_each_side(n_cells_x_0 + n_cells_x_1 + 1)
+        , airfoil_1D(set_airfoil_length(
+            data.airfoil_type == "Jouk" ?
+              this->joukowski(data.center,
+                              N_points_on_each_side,
+                              data.point_factor) :
+              (data.airfoil_type == "NACA" ?
+                 this->naca(data.naca_id,
+                            N_points_on_each_side,
+                            data.point_factor) :
+                 std::array<std::vector<Point<2>>, 2>{
+                   std::vector<Point<2>>{Point<2>(0), Point<2>(1)},
+                   std::vector<Point<2>>{
+                     Point<2>(0),
+                     Point<2>(
+                       1)}} /* dummy vector since we are asserting later*/),
+            data.airfoil_length))
+        , end_b0_x_u(airfoil_1D[0][n_cells_x_0](0))
+        , end_b0_x_l(airfoil_1D[1][n_cells_x_0](0))
+        , nose_x(airfoil_1D[0].front()(0))
+        , tail_x(airfoil_1D[0].back()(0))
+        , tail_y(airfoil_1D[0].back()(1))
+        , center_mesh(0.5 * std::abs(end_b0_x_u + end_b0_x_l))
+        , length_b1_x(tail_x - center_mesh)
+        , gamma(
+            std::atan(height / (edge_length + std::abs(nose_x - center_mesh))))
+        , A(nose_x - edge_length, 0)
+        , B(nose_x, 0)
+        // coarse grid has to be symmetric in respect to x-axis to allow
+        // periodic BC and make sure that interpolate() works
+        , C(center_mesh, +std::abs(nose_x - center_mesh) * std::tan(gamma))
+        , D(center_mesh, height)
+        , E(center_mesh, -std::abs(nose_x - center_mesh) * std::tan(gamma))
+        , F(center_mesh, -height)
+        , G(tail_x, height)
+        , H(tail_x, 0)
+        , I(tail_x, -height)
+        , J(tail_x + length_b2, 0)
+        , K(J(0), G(1))
+        , L(J(0), I(1))
+      {
+        Assert(data.airfoil_type == "Jouk" || data.airfoil_type == "NACA",
+               ExcMessage("Unknown airfoil type."));
+      }
+
+      /**
+       * Create a serial triangulation.
+       */
+      void create_triangulation(
+        Triangulation<2> &tria_grid,
+        std::vector<
+          GridTools::PeriodicFacePair<typename Triangulation<2>::cell_iterator>>
+          *periodic_faces) const
+      {
+        this->make_coarse_grid(tria_grid);
+
+        this->set_boundary_ids(tria_grid);
+
+        if (periodic_faces != nullptr)
+          {
+            GridTools::collect_periodic_faces(
+              tria_grid, 5, 4, 1, *periodic_faces);
+            tria_grid.add_periodicity(*periodic_faces);
+          }
+
+        tria_grid.refine_global(refinements);
+        this->interpolate(tria_grid);
+      }
+
+      /**
+       * Create a parallel::distributed::Triangulation.
+       */
+      void create_triangulation(
+        parallel::distributed::Triangulation<2> &parallel_grid,
```

Comment:
so we can remove  function `void create_triangulation(parallel::distributed::Triangulation<2> &parallel_grid, ...)` since `void create_triangulation(Triangulation<2> &tria_grid, ...)` will work for serial and distributed triangulations, right?

Reply:
I think it will work. You should try it out.